ovh/resource_cloud_project_user.go

```diff
@@ -107,6 +106,28 @@ func resourceCloudProjectUser() *schema.Resource {
 		},
 	}
 }
+func validateCloudProjectUserRoleFunc(config *Config, serviceName string, roles []string, role string) (*CloudProjectrolesResponse, error) {
+
+	endpoint := fmt.Sprintf("/cloud/project/%s/role",
+		url.PathEscape(serviceName),
+	)
+	res := &CloudProjectrolesResponse{}
+	if err := config.OVHClient.Get(endpoint, res); err != nil {
+		return nil, fmt.Errorf("calling Get %s", endpoint)
+	}
+
+	ovhRole := make([]string, len(res.Roles)+1)
```

Comment:
```suggestion
	ovhRole := make([]string, 0, len(res.Roles))
```

or else you will have `len(res.Roles)+1` empty values at the beginning of your array